### PR TITLE
Use address autocomplete

### PR DESCRIPTION
**Pull request by nicksellen**
## What does this PR do?

Use autocomplete for address picker again as it used to. I made it so it gives you an explicit option to use the search term as the address without adding/changing any existing map marker.

Also defaults place edit map to be group centre if available.

I didn't add everything into translation strings, just as maybe there are some language changes in review (plus, I'm tired and about to go to bed ;) )

## Checklist

- [ ] added a test, or explain why one is not needed/possible...
- [ ] no unrelated changes
- [ ] asked someone for a code review
- [ ] joined #karrot-dev channel at https://slackin.yunity.org
- [ ] added an entry to CHANGELOG.md (description, pull request link, username(s))

### CODE REVIEW

File: src/maps/components/AddressPicker.vue [hunk 1]
```diff
@@ -192,7 +197,7 @@ export default {
 <style scoped lang="stylus">
 .map
   height 260px
-  margin-left 42px
+  margin-left 37px
```
Comment (nicksellen) [inline src/maps/components/AddressPicker.vue]: This makes it align nicely for place edit and user profile edit, but not group edit :/

File: src/maps/components/AddressPicker.vue [hunk 2]
```diff
@@ -1,66 +1,64 @@
...
-            <QItemSection>
-              <QItemLabel>
-                {{ $t('GLOBAL.SEARCH_NOT_FOUND') }}
-              </QItemLabel>
-            </QItemSection>
-          </QItem>
-        </QList>
-      </QMenu>
-    </QInput>
+      <template #selected-item />
```
Comment (nicksellen) [inline src/maps/components/AddressPicker.vue]: It's a bit odd that it shows the selected item, as well as the text, maybe there is a better way to fix it up, but did this for now - basically an empty slot template thing.

File: src/maps/components/AddressPicker.vue [hunk 3]
```diff
@@ -1,66 +1,64 @@
...
+          :key="index"
+          v-bind="itemProps"
+          v-on="itemEvents"
+        >
+          <QItemSection>
+            <QItemLabel v-if="useSearchTerm">
+              Set address as
+              <strong>{{ itemLabel }}</strong>
+              <span v-if="hasLocation">(keep existing location on map)</span>
+              <span v-else>(choose location manually on map)</span>
```
Comment (nicksellen) [inline src/maps/components/AddressPicker.vue]: These are the strings I didn't add into translations yet, not sure about the wording at this moment.

**Comment by nicksellen** [comment]
Test failures were a bit weird, unrelated to my changes, but fixed now.

**Comment by tiltec** [comment]
I had a quick try, UI seems good; definitely better than before. Will have a detailed look later into code and wording!

File: src/maps/components/AddressPicker.vue (hunk 1 above)
Comment (tiltec) [reply src/maps/components/AddressPicker.vue]: Seems fine to me...

File: src/maps/components/AddressPicker.vue [hunk 4]
```diff
@@ -1,66 +1,64 @@
...
+            </QItemLabel>
+          </QItemSection>
+        </QItem>
+      </template>
+    </QSelect>
     <StandardMap
       class="map"
       :markers="marker ? [marker] : []"
       :prevent-zoom="preventZoom"
+      :default-center="defaultMapCentre"
```
Comment (tiltec) [inline src/maps/components/AddressPicker.vue]: `defaultMapCenter` seems more consistent

File: src/maps/components/AddressPicker.vue [hunk 5]
```diff
@@ -1,66 +1,64 @@
...
+              <strong>{{ itemLabel }}</strong>
+              <span v-if="hasLocation">({{ $t('ADDRESS_PICKER.KEEP_EXISTING_LOCATION') }})</span>
+              <span v-else>({{ $t('ADDRESS_PICKER.CHOOSE_LOCATION') }})</span>
+            </QItemLabel>
+            <QItemLabel v-else>
+              {{ itemLabel }}
+            </QItemLabel>
+          </QItemSection>
+        </QItem>
+        <QSeparator v-if="useSearchTerm" />
```
Comment (tiltec) [inline src/maps/components/AddressPicker.vue]: Maybe move everything below inside a `<template v-if="useSearchTearm">` to make the meaning more obvious/less repetitive.

**Comment by nicksellen** [comment]
> Aside, I wonder if we should rewrite v-slot: to # to keep it uniform?

Yes, I would like that, seems neater and consistent with generally using short versions where possible.

**Pull request merged**